infoschema/metricschema/table.go

```diff
@@ -105,6 +102,45 @@ func (def *MetricTableDef) GenPromQL(sctx sessionctx.Context, labels []string) s
 	return promQL
 }
 
+func (def *MetricTableDef) genLabelCondition(labels map[string]set.StringSet) string {
+	var buf bytes.Buffer
+	index := 0
+	for _, label := range def.Labels {
+		values := labels[label]
+		if len(values) == 0 {
+			continue
+		}
+		if index > 0 {
+			buf.WriteByte(',')
+		}
+		switch len(values) {
+		case 1:
+			buf.WriteString(fmt.Sprintf("%s=\"%s\"", label, GenLabelConditionValues(values)))
+		default:
+			buf.WriteString(fmt.Sprintf("%s=~\"%s\"", label, GenLabelConditionValues(values)))
+		}
+		index++
+	}
+	return buf.String()
+}
+
+// GenLabelConditionValues generates the label condition values.
+func GenLabelConditionValues(values set.StringSet) string {
+	var buf bytes.Buffer
+	vs := make([]string, 0, len(values))
+	for k := range values {
+		vs = append(vs, k)
+	}
+	sort.Strings(vs)
+	for i, value := range vs {
+		if i > 0 {
+			buf.WriteByte('|')
+		}
+		buf.WriteString(value)
+	}
+	return buf.String()
+}
+
```

Comment:
Prefer to move this function to `executor` package.

Reply:
Move to `executor ` will have circle import problem.